build/compiler.js

```diff
@@ -105,6 +105,19 @@ function getConfig({target, env, dir, watch, cover}) {
   // 'test' and 'production' entries should both map to NODE_ENV='production'
   const nodeEnv = env === 'development' ? 'development' : 'production';
 
+  // Allow overrides with a warning for `dev` and `test` commands. In production builds, throw if NODE_ENV is not `production`.
+  const nodeEnvBanner =
+    `if(process.env.NODE_ENV && process.env.NODE_ENV !== '${nodeEnv}') {` +
+    `if ('${env}' === 'production') {` +
```

Comment:
The logic here is sound, but I wonder if there's any (minuscule) benefit of pulling this particular `if/else` out of the template since we can resolve these values earlier at build time.